Show current timer status of Scotte burners and display it in system image.

syastrov opened the pull request:
So, I thought it would be useful to have an overview of what is going on on the system image when using the Scotte built-in timer to start/stop the burner.

The plugin sets the item called `timer_status` with a string, which is calculated based on `time_minutes` and `timer_heating_period` and the `timer_heating__start_X` settings.

If it is not currently within a timer period, then it determines when the burner is set to start ("Starting at ...").
Otherwise, it tries to determine when the next time the burner is set to stop according to the timer ("Stopping at ...").
If period is not set (0), then it says "Timer disabled".

The value has been added to the default system image.
The plugin has been enabled by default (don't know if that's a good idea).

<img width="593" alt="screen shot 2018-05-16 at 23 06 11" src="https://user-images.githubusercontent.com/5605897/40146240-13beb156-5965-11e8-8796-f066d35b82a4.png">

Not sure if it belongs in a separate plugin, but I didn't know where else it should fit in.

syastrov committed:
Add timer_status to Scottecom plugin and display it in the NBE system images.

syastrov commented:
I guess you can say it's a work-in-progress...
It doesn't seem to update if I set the `timer_heating_period` to "0".

motoz commented:
Nice work! That's really appreciated!

The problem you are seeing with time_heating_period = "0" is probably due to db.get_value returning strings, so the comparison to an integer zero is always false.

syastrov commented:
You're right, that was the issue. I updated the PR and fixed a couple of other bugs:
* Would not consider heating start 4 (off-by-one error)
* Would not filter out heating starts set to "00;00" (the control box interprets that as "not set")

It seems to be working now.

motoz commented:
Great!

I'd like to merge this, but there are a couple of things that we should discuss first.

First, the new files you are adding are missing from the autotools build system, so they won't get installed when running 'make install', nor will they be included when making a .deb package. But when thinking about this I actually feel that this could be part of the 'scottecom' plugin instead of being a new plugin, since it's only useful together with that plugin anyway. If you add a file to the scottecom plugin you only need to add it in one makefile template here: https://github.com/motoz/PellMon/blob/master/src/Pellmonsrv/plugins/scottecom/Makefile.am

The second thing is your modification to the default system image. I really like the look of the clock btw... But pellmon has a lot of users with 'generic' burners that are using the 'pelletcalc' plugin instead of scottecom/nbecom, the idea is that the default image should be generic enough for them also. Maybe you could add the timer to one of the NBE V4-5-6 specific system images instead? (system_nbe.svg, system_nbe_2w.svg, system_nbe_3w.svg).

syastrov commented:
Thanks for the feedback!
I've moved the code into the scottecom plugin and added the timer to the NBE V4-5-6 system images, as you suggested.

motoz commented:
Super, I'll pull it right away. Thanks for your contribution!
